ui/components/app/detected-token/detected-token.js

```diff
@@ -43,13 +48,42 @@ const DetectedToken = ({ setShowDetectedTokens }) => {
   ] = useState(false);
 
   const handleClearTokensSelection = async () => {
-    // create a lodash chain on this object
     const {
-      selected: selectedTokens,
-      deselected: deSelectedTokens,
+      selected: selectedTokens = [],
+      deselected: deSelectedTokens = [],
     } = sortingBasedOnTokenSelection(tokensListDetected);
 
+    if (deSelectedTokens.length > 0) {
+      const tokensDetailsList = deSelectedTokens.map(
+        ({ symbol, address }) => `${symbol} - ${address}`,
+      );
+      trackEvent({
+        event: EVENT_NAMES.TOKEN_HIDDEN,
+        category: EVENT.CATEGORIES.WALLET,
+        sensitiveProperties: {
+          tokens: tokensDetailsList,
+          location: EVENT.LOCATION.TOKEN_DETECTION,
+          token_standard: TOKEN_STANDARDS.ERC20,
+          asset_type: ASSET_TYPES.TOKEN,
+        },
+      });
+    }
+
     if (deSelectedTokens.length < detectedTokens.length) {
+      selectedTokens.forEach((importedToken) => {
+        trackEvent({
+          event: EVENT_NAMES.TOKEN_ADDED,
+          category: EVENT.CATEGORIES.WALLET,
+          sensitiveProperties: {
+            token_symbol: importedToken.symbol,
+            token_contract_address: importedToken.address,
+            token_decimal_precision: importedToken.decimals,
+            source: EVENT.SOURCE.TOKEN.DETECTED,
+            token_standard: TOKEN_STANDARDS.ERC20,
+            asset_type: ASSET_TYPES.TOKEN,
+          },
+        });
+      });
       await dispatch(ignoreTokens(deSelectedTokens));
```

Comment:
NIT: kinda out of the scope of this PR, but I think we could remove the `else` statement and move `await dispatch(ignoreTokens(deSelectedTokens));` underneath